### PR TITLE
forms-provider, adjust failedRule behaviour

### PR DESCRIPTION
**Pull request by ArnoBuschmann**
Currently failedRule is still set, even if it is valid. With this code change it gets set to null. As I am a JS beginner, I would appreciate, if someone else could write a test, thanks :)

### CODE REVIEW

**Comment by ArnoBuschmann** [comment]
@henri-hulski arrgh yeah, that line should stay unchanged. It dropped in because I played around in the version that has been built for the node package alread, if that makes sense?

**Comment by henri-hulski** [comment]
I see. :-)

**Comment by ArnoBuschmann** [comment]
@henri-hulski just changed that. Resulted in a second commit, being slightly unsure if this was the correct way how to do it as this is my first PR in general ;)

**Comment by edgesoft** [comment]
@ArnoBuschmann Great to see your first PR. As you can see the build is failing. It's linting. you can check it by running npm run lint and make adjustments. Can you add a test there as well? :)

File: packages/cerebral-provider-forms/src/utils/validate.js [hunk 1]
```diff
@@ -34,11 +34,14 @@ export default function validate (form, value, validationRules) {
...
+    const isValid = rule(value, form, arg)
+
     return {
-      isValid: rule(value, form, arg),
-      failedRule: {
+      isValid,
+      failedRule: isValid ? null : {
         name: ruleKey,
-        arg
+        arg: arg
```
Comment (henri-hulski) [inline packages/cerebral-provider-forms/src/utils/validate.js]: `arg` is the same as `arg: arg`. 
So you can leave it.

**Comment by ArnoBuschmann** [comment]
@edgesoft thanks for the hints! Tehe, never have been writing tests. But sure, let's get into this as well, I will see, what I can do :)

File: packages/cerebral-provider-forms/src/utils/validate.js (hunk 1 above)
Comment (ArnoBuschmann) [inline packages/cerebral-provider-forms/src/utils/validate.js]: Indeed, will change that next, together with the needed lint changes.

File: packages/cerebral-provider-forms/src/utils/validate.js [hunk 2]
```diff
@@ -34,9 +34,12 @@ export default function validate (form, value, validationRules) {
       arg = undefined
     }
 
+
```
Comment (Guria) [inline packages/cerebral-provider-forms/src/utils/validate.js]: extra empty line

Comment (ArnoBuschmann) [inline packages/cerebral-provider-forms/src/utils/validate.js]: @Guria thanks, whoa bear with me :)

**Comment by edgesoft** [comment]
@ArnoBuschmann Still linting validate.js:36

**Comment by edgesoft** [comment]
@ArnoBuschmann Sorry! Saw in your first post that you wanted someone else write a test. Will fix!

**Comment by ArnoBuschmann** [comment]
@edgesoft Always great to lean new things, so I will have a look at how to write tests. I'm just not sure yet, if I could provide this one quickly ;)

**Comment by Guria** [comment]
@ArnoBuschmann just add some test cases into https://github.com/ArnoBuschmann/cerebral/blob/e5970de2f6400f0cbc68084cea3f80c1b9642a78/packages/cerebral-provider-forms/src/util.test.js

You just describe what are you expecting to get from function with predefined input and check it with assertions.

**Comment by ArnoBuschmann** [comment]
@Guria thanks a lot for the explanation! Will have a look at it and give it a try, later today or early tomorrow.

File: packages/cerebral-provider-forms/src/utils/validate.js [hunk 3]
```diff
@@ -34,13 +34,16 @@ export default function validate (form, value, validationRules) {
       arg = undefined
     }
 
+    
+    const isValid = rule(value, form, arg)
+
     return {
```
Comment (yusufsafak) [inline packages/cerebral-provider-forms/src/utils/validate.js]: Hi @ArnoBuschmann, 
If field is valid, I think it is better to return initialValidation.
 `return isValid ? initialValidation : {
  .....
}`

**Pull request merged**